internal/driver/virtualdriver.go

```diff
@@ -83,7 +83,15 @@ func (d *VirtualDriver) HandleReadCommands(deviceName string, protocols map[stri
 
 	for i, req := range reqs {
 		if dr, ok := d.sdk.DeviceResource(deviceName, req.DeviceResourceName); ok {
-			if v, err := vd.read(deviceName, req.DeviceResourceName, dr.Properties.ValueType, dr.Properties.Minimum, dr.Properties.Maximum, d.db); err != nil {
+			min := float64(0)
+			if dr.Properties.Minimum != nil {
+				min = *dr.Properties.Minimum
+			}
+			max := float64(0)
+			if dr.Properties.Maximum != nil {
+				min = *dr.Properties.Maximum
+			}
```

Comment:
don't give the default value here.  0 is not a reasonable default value.

Reply:
Previously, if it's empty string or not set, it will cause the parsing error in the driver.  Then, the max and min of that data type will be applied to generate the random values.

https://github.com/edgexfoundry/device-virtual-go/blob/b76ffb3f756266f027a45c1155223ddd586b2bad/internal/driver/resourceint.go#L36-L45

Now, if they are not set, in golang, it will be nil, not an empty string.  I am asking Bruce to throw the parsing error in the same place if they are nil.

Reply:
Updated, use the default value if the minimum and maximum is nil.